pallets/file-system/src/utils.rs

```diff
@@ -721,143 +683,56 @@ where
-                        .try_push((file_key.0, file_key.1.clone()))
-                        .map_err(|_| Error::<T>::TooManyStorageRequestResponses)?;
-                }
-
-                if !rejected.is_empty() {
-                    let rejected_batch = MspRejectedBatchStorageRequests {
-                        file_keys: rejected,
-                        bucket_id,
-                        owner: owner.clone(),
-                    };
-
-                    batch_responses
-                        .try_push(BatchResponses::Rejected(rejected_batch))
-                        .map_err(|_| Error::<T>::TooManyBatchResponses)?;
-                }
-            }
-
-            if !failed.is_empty() {
-                let failed_batch = MspFailedBatchStorageRequests {
-                    file_keys: failed,
-                    bucket_id,
-                    owner: owner.clone(),
-                };
+                Self::cleanup_storage_request(
+                    EitherAccountIdOrMspId::MspId(msp_id),
+                    file_key,
+                    &storage_request_metadata,
+                )?;
 
-                batch_responses
-                    .try_push(BatchResponses::Failed(failed_batch))
-                    .map_err(|_| Error::<T>::TooManyBatchResponses)?;
+                Self::deposit_event(Event::StorageRequestRejected {
+                    file_key: file_key,
+                    reason,
+                });
             }
         }
 
-        // Construct the result
-        let result = MspRespondStorageRequestsResult {
-            msp_id,
-            responses: batch_responses,
-        };
-
-        Ok(result)
+        Ok(())
     }
 
-    /// Accept as many storage requests as possible (best-effort) belonging to the same bucket.
+    /// Accept all storage requests belonging to the same bucket in an all-or-nothing operation.
     ///
     /// There should be a single non-inclusion forest proof for all file keys, and finally there should
     /// be a list of file key(s) with a key proof for each of them.
     ///
     /// The implementation follows this sequence:
     /// 1. Verify the non-inclusion proof.
-    /// 2. For each file key: Record a successful acceptance or a failure. Any failed operation while processing a file key
-    /// will not result in the function failing, but the file key will be marked as failed and the function will continue processing the rest.
-    /// 3. Apply the delta with all the keys that were successfully accepted to the root of the bucket.
+    /// 2. For each file key: Verify and process the acceptance. If any operation fails during the processing of a file key,
+    /// the entire function will fail and no changes will be applied.
+    /// 3. If all file keys are successfully processed, apply the delta with all the accepted keys to the root of the bucket.
+    /// 4. If any step fails, the function will return an error and no changes will be made to the storage state.
```

Comment:
There's a caveat to this sentence. Actually, no changes will be applied to storage, so long as this function is called within an extrinsic that propagates the error and makes the extrinsic fail. Only if the overarching extrinsic fails, then the storage is not updated.